packages/inter-protocol/src/auction/auctionBook.js

```diff
@@ -0,0 +1,339 @@
+ * orders:
+ *   - Prices express the bid in terms of a Currency amount
+ *   - Scaled bid  express the bid in terms of a discount (or markup) from the
+ *     most recent oracle price.
+ *
+ * Offers can be added in three ways. When the auction is not active, prices are
+ * automatically added to the appropriate collection. If a new offer is at or
+ * above the current price of an active auction, it will be settled immediately.
+ * If the offer is below the current price, it will be added, and settled when
+ * the price reaches that level.
+ */
+
+const trace = makeTracer('AucBook', false);
+
+/** @typedef {import('@agoric/vat-data').Baggage} Baggage */
+
+export const makeAuctionBook = async (
+  baggage,
+  zcf,
+  currencyBrand,
+  collateralBrand,
+  priceAuthority,
+) => {
+  const zeroRatio = makeRatioFromAmounts(
+    AmountMath.makeEmpty(currencyBrand),
+    AmountMath.make(collateralBrand, 1n),
+  );
+  const BidSpecShape = M.or(
+    {
+      want: AmountShape,
+      offerPrice: makeBrandedRatioPattern(currencyBrand, collateralBrand),
+    },
+    {
+      want: AmountShape,
+      offerBidScaling: makeBrandedRatioPattern(currencyBrand, currencyBrand),
+    },
+  );
+
+  let assetsForSale = AmountMath.makeEmpty(collateralBrand);
+  const { zcfSeat: collateralSeat } = zcf.makeEmptySeatKit();
+  const { zcfSeat: currencySeat } = zcf.makeEmptySeatKit();
+
+  let lockedPriceForRound = zeroRatio;
+  let updatingOracleQuote = zeroRatio;
+  E.when(E(collateralBrand).getDisplayInfo(), ({ decimalPlaces = 9n }) => {
+    // TODO(#6946) use this to keep a current price that can be published in state.
+    const quoteNotifier = E(priceAuthority).makeQuoteNotifier(
+      AmountMath.make(collateralBrand, 10n ** decimalPlaces),
+      currencyBrand,
+    );
+
+    observeNotifier(quoteNotifier, {
+      updateState: quote => {
+        trace(
+          `BOOK notifier ${priceFrom(quote).numerator.value}/${
+            priceFrom(quote).denominator.value
+          }`,
+        );
+        return (updatingOracleQuote = priceFrom(quote));
+      },
+      fail: reason => {
+        throw Error(`auction observer of ${collateralBrand} failed: ${reason}`);
+      },
+      finish: done => {
+        throw Error(`auction observer for ${collateralBrand} died: ${done}`);
+      },
+    });
+  });
+
+  let curAuctionPrice = zeroRatio;
+
+  const scaledBidBook = provide(baggage, 'scaledBidBook', () => {
+    const scaledBidStore = makeScalarBigMapStore('scaledBidBookStore', {
+      durable: true,
+    });
+    return makeScaledBidBook(scaledBidStore, currencyBrand, collateralBrand);
+  });
+
+  const priceBook = provide(baggage, 'sortedOffers', () => {
+    const priceStore = makeScalarBigMapStore('sortedOffersStore', {
+      durable: true,
+    });
+    return makePriceBook(priceStore, currencyBrand, collateralBrand);
+  });
+
+  const removeFromOneBook = (isPriceBook, key) => {
+    if (isPriceBook) {
```

Comment:
this isPriceBook signal is loose. looks like it's receiving a price object, so please declare that as the param. 
```suggestion
  /** @type { (price: Price | undefined, key: string} */ 
  const removeFromOneBook = (price, key) => {
    if (price) {
```
Even better put `key` first and make `price` optional.

Reply:
done. (both)